daml-lf/spec/daml-lf-1.rst

```diff
@@ -940,17 +994,17 @@ Then we define *well-formed expressions*. ::
     ——————————————————————————————————————————————————————————————— ExpGenMap (*)
       Γ  ⊢  〚e₁ ↦ e₁'; …; eₙ ↦ eₙ'〛: GenMap σ τ
 
-      τ contains no quantifiers
+      τ contains no quantifiers nor type synonyms
       ε  ⊢  τ : *     Γ  ⊢  e  : τ
     ——————————————————————————————————————————————————————————————— ExpToAny
       Γ  ⊢  'to_any' @τ e  :  'Any'
 
-      τ contains no quantifiers
+      τ contains no quantifiers nor type synonyms
       ε  ⊢  τ : *     Γ  ⊢  e  : Any
```

Comment:
Why not?

```suggestion
      τ' contains no quantifiers nor type synonyms
      τ ↠ τ'       ε  ⊢  τ' : *     Γ  ⊢  e  : Any
```

Reply:
It’s a bit annoying to implement since you would need to expand it in the speedy compiler so given that we don’t need it, it seems reasonable to simply forbid it at least for now.

Reply:
I see, that seems reasonable. The idea is that any user-defined type synonyms DAML will be expanded already in DAML-LF, and any type synonym in DAML-LF will be from constraints/typeclasses in DAML, which we don't care to marshal in or out of `Any`.